packages/test/src/index.ts

```diff
@@ -178,3 +180,11 @@ export const otel = Object.freeze({
   /** Reset OTel, optionally with a custom config */
   reset: (config?: OtelConfig) => resetOtel(config),
 })
+
+beforeEach(async function () {
+  const CURRENT_RETRY = "currentRetry"
+  const currentRetry = this.currentTest?.[CURRENT_RETRY]()
+  if (currentRetry) {
+    await setTimeout(currentRetry * 1000)
+  }
+})
```

Comment:
Introduce a timeout between failed test retries (useful for sampler integration tests where we want to get a successful sampling decision but are hitting a rate limit)